src/analyses/variable-sensitivity/abstract_environment.h

```diff
@@ -96,6 +96,7 @@ class abstract_environmentt
   /// The better the implementation the more precise the results
   /// will be.
   virtual bool assume(const exprt &expr, const namespacet &ns);
+  exprt do_assume(const exprt &e, const namespacet &ns);
```

Comment:
Is there any harm in this being public?  I don't think so but I thought I should check.

Reply:
It's public so I can exercise it directly in unit tests. (Rather than via `assume` which does some other things too.) However, there's no harm can come of it been public.

Reply:
That is fine.